lib/Service/PdfService.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Richdocuments\Service;
+
+use mikehaertl\pdftk\Pdf;
+use OCP\Files\Node;
+use OCP\IConfig;
+use Psr\Log\LoggerInterface;
+
+class PdfService {
+	public function __construct(
+		private IConfig $config,
+		private LoggerInterface $logger
+	) {
+	}
+
+	public function extractFields(Node $file): array {
+		$filePath = $file->getStorage()->getLocalFile($file->getInternalPath());
+
+		try {
+			$pdf = new Pdf($filePath);
+			$fields = $pdf->getDataFields();
+			$templateFields = [];
+			foreach ($fields as $field) {
+				$templateFields[] = [
+					'name' => $field['FieldName'],
+					'type' => $field['FieldType'],
+					'default' => $field['FieldValueDefault'] ?? '',
+					'options' => $field['FieldStateOption'] ?? [],
+				];
```

Comment:
@luka-nextcloud This needs a few adjustments after rebasing to the latest state of the PR from @elzody 

There is now a dedicated Field class that can be used for this.

Reply:
@juliushaertl Which class do you mean?

Reply:
There is a Field class in server now that can be used for this. Checkout the implementation around https://github.com/nextcloud/richdocuments/blob/6f85ff4ff176870a0f086267b267c21a97544c74/lib/Service/TemplateFieldService.php#L63-L70 for office file types